src/software/amazon/ion/IonReader.java

```diff
@@ -172,6 +172,17 @@
     public IonType getType();
 
     /**
+     * Returns the actual type of the Ion <code>int</code> at the current
+     * value.
+     *
+     * @return an {@link IntegerType} denoting the current <code>int</code>
+     * value's Java type, or <code>null</code> if the current value is
+     * <code>null.int</code>.
+     * @see IonInt#getIntegerType()
+     */
+    public IntegerType getIntegerType();
```

Comment:
Eliminated `IntegerType` altogether as part of the refactor into a callback API. There will now be no difference between the latter three states -- they all trigger the same callback.